task/run.go

```diff
@@ -17,6 +17,136 @@ import (
 
 const trueString = "true"
 
+// RunWithJSONOutput orders the tasks be dependencies to build an execution plan and then executes each required task,
+// and writes all log lines as JSON.
+func RunWithJSONOutput(registry *Registry, arguments []string) error {
```

Comment:
Not sure if theres a better name, or place to put this.

Its also not super DRY, since it repeats most of `Run`, but there wasn't a super clear way to re-use code here.